File: articles/desalination-plant-security/cpp/service_continuity.cpp
#include <iostream>

bool meets_priority_demand(
    double desalination_output,
    double outage_fraction,
    double alternative_supply,
    double emergency_transfer,
    double losses,
    double priority_demand
) {
    const double outage_supply = ((1.0 - outage_fraction) * desalination_output)
                               + alternative_supply
                               + emergency_transfer
                               - losses;
    return outage_supply >= priority_demand;
}

int main() {
    const bool stable = meets_priority_demand(420.0, 0.25, 85.0, 40.0, 28.0, 360.0);
    std::cout << "Synthetic system meets priority demand under outage: "
              << (stable ? "true" : "false") << "\n";
    return 0;
}

File: articles/desalination-plant-security/fortran/service_continuity.f90
program service_continuity
  implicit none

  real :: desalination_output, outage_fraction, alternative_supply
  real :: emergency_transfer, losses, priority_demand, outage_supply
  logical :: stable

  desalination_output = 420.0
  outage_fraction = 0.25
  alternative_supply = 85.0
  emergency_transfer = 40.0
  losses = 28.0
  priority_demand = 360.0

  outage_supply = ((1.0 - outage_fraction) * desalination_output) + &
                  alternative_supply + emergency_transfer - losses

  stable = outage_supply >= priority_demand

  print *, "Synthetic system meets priority demand under outage:", stable
end program service_continuity
